instrumentation/jdbc/metadata.yaml

```diff
@@ -1,16 +1,25 @@
-disabled_by_default: true
 description: >
   The JDBC instrumentation provides database client spans and metrics. Each call produces a span
   named after the SQL verb, enriched with standard DB client attributes (system, database,
   operation, sanitized statement, peer address) and error details if an exception occurs.
+  
+  There is also a "jdbc-datasource" instrumentation that creates spans for datasource connections,
+  but is disabled by default due to the volume of telemetry produced.
 configurations:
   - name: otel.instrumentation.jdbc.statement-sanitizer.enabled
     description: Enables statement sanitization for database queries. Takes precedent to 
       otel.instrumentation.common.db-statement-sanitizer.enabled.
+    type: boolean
     default: true
   - name: otel.instrumentation.common.db-statement-sanitizer.enabled
     description: Enables statement sanitization for database queries.
+    type: boolean
     default: true
   - name: otel.instrumentation.common.peer-service-mapping
     description: Used to specify a mapping from host names or IP addresses to peer services.
+    type: map
     default: ""
+  - name: otel.instrumentation.jdbc-datasource.enabled
```

Comment:
An alternative to this would be to allow metadata.yaml to describe multiple instrumentations. I guess it is tricky, in some cases on module contains multiple instrumentation modules that operate separately like here the jdbc and datasource instrumentation or kafka and kafka-metrics while in other cases multiple modules could be merged like opensearch-rest-1.0 and opensearch-rest-3.0.

Reply:
I will do some thinking on how we can better handle these types of situations